### PR TITLE
Simulator SPI

### PR DESCRIPTION
**Pull request by manthanand**

### CODE REVIEW

File: BSP/Simulator/Src/BSP_SPI.c [hunk 1]
```diff
@@ -229,11 +248,13 @@ static void WRCommandHandler(uint8_t *buf, uint32_t len) {
         case SIM_LTC6811_WRCFGA: {
```
Comment (ClarkPoon) [inline BSP/Simulator/Src/BSP_SPI.c:248]: Are you actually using any of the private functions?

File: Drivers/Src/LTC681x.c [hunk 2]
```diff
@@ -48,17 +48,17 @@ Copyright 2017 Linear Technology Corp. (LTC)
 
 static uint8_t spi_read8(void){
     uint8_t data = 0;
```
Comment (ClarkPoon) [inline Drivers/Src/LTC681x.c:50]: You will need to uncomment these lines before we can merge into simulator

**Review by ClarkPoon — CHANGES_REQUESTED** [review]

File: BSP/Simulator/DataGeneration/SPI.py [hunk 3]
```diff
@@ -16,6 +16,7 @@
 wires = []                  # list of 31 modules (1 = connected; 0 = open)
 voltage_values = []         # list of 31 modules (fixed point 0.0001)
 temperature_values = []     # list of 31 pairs of sensors (fixed point 0.001)
+DCC_values = [0] * 31             # list of 31 modules (1 = discharging)
```
[the remark this answers is not in the document]
Comment (manthanand) [reply BSP/Simulator/DataGeneration/SPI.py:19]: No, I deleted it

[the remark this answers is not in the document]
Comment (manthanand) [reply BSP/Simulator/DataGeneration/SPI.py]: For some reason, the code won't run without this

File: BSP/Simulator/Src/BSP_SPI.c (hunk 1 above)
Comment (manthanand) [reply BSP/Simulator/Src/BSP_SPI.c:246]: I commented out all the private functions and it works

File: BSP/Inc/BSP_SPI.h [hunk 4]
```diff
@@ -24,23 +24,21 @@ void BSP_SPI_Init(void);
  *          the SPI protocol expects where a transmit and receive happen
  *          simultaneously.
  * @note    Blocking statement
- * @param   txBuf   data array that contains the data to be sent.
- * @param   txLen   length of data array.
+ * @param   data  memory address to data array that contains the data to be sent.
  * @return  None
  */
-void BSP_SPI_Write(uint8_t *txBuf, uint32_t txLen);
```
[the remark this answers is not in the document]
Comment (ClarkPoon) [reply BSP/Inc/BSP_SPI.h:31]: Can you also add back the comments that you deleted?

**Review by ClarkPoon — COMMENTED** [review]
You are missing some comments that you deleted in BSP_SPI.h. Other than that, I think this is readyto test with Sugam's code

**Comment by chaseblock** [comment]
What's the status on this?

**Comment by ClarkPoon** [comment]
> What's the status on this?

I think @SijWoo is fixing this issue on a different branch. I don't want to get rid of this until his fix is merged.

**Comment by SijWoo** [comment]
I'm going to close this because this is now obsolete and open a new PR for simulator_ltc6811

**Pull request closed without merging**